zrml/prediction-markets/src/lib.rs

```diff
@@ -135,21 +146,45 @@ mod pallet {
             origin: OriginFor<T>,
             market_id: MarketIdOf<T>,
         ) -> DispatchResultWithPostInfo {
-            T::DestroyOrigin::ensure_origin(origin)?;
+            ensure_root(origin)?;
 
             let mut total_accounts = 0usize;
             let mut share_accounts = 0usize;
             let market = T::MarketCommons::market(&market_id)?;
             let market_status = market.status;
             let outcome_assets = Self::outcome_assets(market_id, &market);
             let outcome_assets_amount = outcome_assets.len();
+
+            // Slash outstanding bonds
+            if market_status == MarketStatus::Proposed {
+                CurrencyOf::<T>::slash_reserved_named(
+                    &RESERVE_ID,
+                    &market.creator,
+                    T::AdvisoryBond::get(),
+                );
+            }
+            if market_status != MarketStatus::Resolved {
+                if market.creation == MarketCreation::Permissionless {
```

Comment:
Yeah it makes sense to only slash the `ValidityBond` in permissionless mode, but why do you exclude the `resolved` market status, when it is only required to slash from `proposed` to `active` and from `proposed` to `CollectingSubsidy`. 

I would expect, that this code slashes too often.

Reply:
The `ValidityBond` is reserved when a permissionless market is created. _Permissionless_ refers to the market creation. If the market moves through its lifecycle as expected, the `ValidityBond` is unreserved when the market is resolved.

If the market is now unexpectedly destroyed mid-usage, the `ValidityBond` must be slashed, but only if it hasn't been unreserved already, i.e. only if the market is not yet resolved.